### ISSUE TITLE
Full cluster restore breaks user privileges

### ISSUE THREAD

**Issue by jfrconley** [cockroachdb/cockroach#93298]
**Describe the problem**

Full cluster restore does not properly restore user permissions

**To Reproduce**

1. Create a user and grant them admin privileges
1. Perform a full backup
1. Create a new cluster and restore from backup
1. Attempt to access admin only resource (eg: SELECT * FROM system.users;)

**Expected behavior**

Access to restricted resources should be allowed

**Additional data / screenshots**
Some permissions are correctly restored, but not all.
For example, SELECT statements on tables in public schemas seem to work fine.
However, special permissions like creating changefeeds are still not allowed.

Additionally, some users do not seem to have their passwords restored correctly.
This results in password auth errors when logging in.

**Environment:**
- CockroachDB version: v22.1.8
- Server OS: Linux
- Client app: CockroachDB SQL Client, Datagrip

**Additional context**
Full cluster restore effectively breaks access to our database, requiring multiple manual steps to restore access.

Jira issue: CRDB-22262

Epic CRDB-21944

**Comment by amruss** [comment]
@livlobo removing it from out backlog let me know if you think otherwise

**Comment by rafiss** [comment]
This seems related to https://github.com/cockroachdb/cockroach/issues/45359

**Comment by adityamaru** [comment]
Hi @jfrconley! Thanks for the bug report, I plan to dig into this over the next couple of days. I'll keep you posted about what I find.

**Comment by Bazaar1** [comment]
HI @adityamaru, did you get the chance to take a look at this?

**Comment by adityamaru** [comment]
Hey apologies for the delay on this it completely fell off my radar. The issue appears to have been fixed on 22.2+ versions of cockroach. Here are screenshots of the reproduction steps on v22.2.11.

> Backup

<img width="872" alt="Screenshot 2023-09-13 at 8 54 28 AM" src="https://github.com/cockroachdb/cockroach/assets/13837382/dafe0d0d-066c-4f52-aa46-c056064094e3">


> Restore
<img width="878" alt="Screenshot 2023-09-13 at 8 55 31 AM" src="https://github.com/cockroachdb/cockroach/assets/13837382/1d7238fa-25fd-4035-bc1f-3829befd91f9">

Since 22.1 is no longer publishing dot releases, can I request you to upgrade and see if that solves your issue too?

**Comment by adityamaru** [comment]
In general a cluster restore is expected to bring back all the permissions in the backed up cluster. It is possible there was a bug in 22.1 that has since been fixed in later versions. I have also tested:

- GRANT CHANGEFEED ON TABLE example_table TO user; and ensured this privilege comes back on cluster restore.
- Create a user with a password and ensure they are restored correctly with their password.

If you are running into issues on 22.2+ versions of cockroach can i request you to provide updated reproduction steps?
